### PR TITLE
Gutenboarding: Fix minor type issues

### PR DESCRIPTION
**Pull request by sirreal**
#### Changes proposed in this Pull Request

Fixes several type issues. Prep for #37945.

#### Testing instructions

`npm run typecheck -- -p client/landing/gutenboarding` shows fewer errors than on `master`. 4 at the moment.

### CODE REVIEW

File: client/landing/gutenboarding/components/domain-picker/button.tsx [hunk 1]
```diff
@@ -27,11 +27,11 @@ const DomainPickerButton: FunctionComponent< Props > = ( {
 	queryParameters,
 	...buttonProps
 } ) => {
-	const buttonRef = createRef();
+	const buttonRef = createRef< HTMLButtonElement >();
 
 	const [ isDomainPopoverVisible, setDomainPopoverVisibility ] = useState( false );
 
-	const handleClose = ( e?: FocusEvent ) => {
+	const handleClose = ( e?: React.FocusEvent ) => {
```
Comment (ockham) [inline client/landing/gutenboarding/components/domain-picker/button.tsx]: Alternatively, can't we just drop our [`onFocusOutside`](https://github.com/Automattic/wp-calypso/pull/38093/files#diff-d110ec85e6d318fd110d71ccd37806cbR64) from `Popover`?

Asking b/c https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ff69d22d5d124148da0a034711c573cf044f6a82/types/wordpress__components/popover/index.d.ts#L82

File: client/landing/gutenboarding/onboarding-block/vertical-select/index.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ import { STORE_KEY } from '../../stores/onboard';
 import { SiteVertical, isFilledFormValue } from '../../stores/onboard/types';
 import { InjectedStepProps } from '../stepper-wizard';
 import Question from '../question';
-import { __TodoAny__ } from 'client/types';
+import { __TodoAny__ } from '../../../../types';
```
Comment (ockham) [inline client/landing/gutenboarding/onboarding-block/vertical-select/index.tsx]: Was thinking those could make sense in a reusable package maybe.

File: client/landing/gutenboarding/stores/onboard/types.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export interface SiteVertical {
 export const EMPTY_FORM_VALUE = Object.freeze( { __EFV__: '__EFV__' } );
 export type FormValue< T > = T | typeof EMPTY_FORM_VALUE;
 export function isFilledFormValue< T >( value: FormValue< T > ): value is T {
-	return value !== EMPTY_FORM_VALUE && ! isShallowEqual( value, EMPTY_FORM_VALUE );
+	return value !== EMPTY_FORM_VALUE && ! isShallowEqual( value as object, EMPTY_FORM_VALUE );
```
Comment (ockham) [inline client/landing/gutenboarding/stores/onboard/types.ts]: Related: #38109

File: client/landing/gutenboarding/components/domain-picker/button.tsx (hunk 1 above)
Comment (sirreal) [reply client/landing/gutenboarding/components/domain-picker/button.tsx]: That was required to handled when the click outside is _on the toggle that closes_ the popover. That would trigger the tuggle twice and result in the popover not being closed.

**Pull request merged**